tests/rados/test_bluestoretool_workflows.py

```diff
@@ -32,155 +36,288 @@ def run(ceph_cluster, **kw):
-    - ceph-bluestore-tool bluefs-bdev-migrate --path osd path --dev-target new-device --devs-source device1
     - ceph-bluestore-tool restore_cfb --path osd path
+    - ceph-bluestore-tool bluefs-bdev-migrate --path osd path --dev-target new-device --devs-source device1
     """
     log.info(run.__doc__)
     config = kw["config"]
     rhbuild = config.get("rhbuild")
     cephadm = CephAdmin(cluster=ceph_cluster, **config)
     rados_obj = RadosOrchestrator(node=cephadm)
     bluestore_obj = BluestoreToolWorkflows(node=cephadm)
+    client = ceph_cluster.get_nodes(role="client")[0]
 
     out, _ = cephadm.shell(args=["ceph osd ls"])
     osd_list = out.strip().split("\n")
 
     try:
-        # Execute ceph-bluestore-tool --help
-        osd_id = random.choice(osd_list)
-        log.info(
-            f"\n --------------------"
-            f"\n Running cbt help for OSD {osd_id}"
-            f"\n --------------------"
-        )
-        out = bluestore_obj.help(osd_id=osd_id)
-        log.info(out)
-
-        # Execute ceph-bluestore-tool fsck --path <osd_path>
-        osd_id = random.choice(osd_list)
-        log.info(
-            f"\n --------------------"
-            f"\n Running consistency check for OSD {osd_id}"
-            f"\n --------------------"
-        )
-        out = bluestore_obj.run_consistency_check(osd_id=osd_id)
-        log.info(out)
-        assert "success" in out
-
-        # Execute ceph-bluestore-tool fsck --deep --path <osd_path>
-        osd_id = random.choice(osd_list)
-        log.info(
-            f"\n --------------------"
-            f"\n Running deep consistency check for OSD {osd_id}"
-            f"\n --------------------"
-        )
-        out = bluestore_obj.run_consistency_check(osd_id=osd_id, deep=True)
-        log.info(out)
-        assert "success" in out
-
-        if rhbuild.split(".")[0] >= "6":
-            # Execute ceph-bluestore-tool qfsck --path <osd_path>
+        if config.get("non-collocated"):
+            # Execute ceph-bluestore-tool --help
+            osd_id = random.choice(osd_list)
+            log.info(
+                f"\n ---------------------------------"
+                f"\n Running cbt help for OSD {osd_id}"
+                f"\n ---------------------------------"
+            )
+            out = bluestore_obj.help(osd_id=osd_id)
+            log.info(out)
+
+            # Add a new dedicated WAL device to existing collocated OSD
+            # ceph-bluestore-tool bluefs-bdev-new-wal --path osd path --dev-target new-device
+            osd_id = random.choice(osd_list)
+            log.info(
+                f"\n -------------------------------------------"
+                f"\n Adding a dedicated WAL device for OSD.{osd_id}"
+                f"\n -------------------------------------------"
+            )
+            osd_host = rados_obj.fetch_host_node(daemon_type="osd", daemon_id=osd_id)
+            empty_devices = rados_obj.get_available_devices(
```

Comment:
By @pdhiran : Add a check after this point to ensure empty devices are available and `empty_device` variable is not an empty list